src/Symfony/Component/Serializer/NameConverter/CamelCaseToSnakeCaseNameConverter.php

```diff
@@ -48,7 +48,7 @@ public function normalize($propertyName)
 
             $len = strlen($propertyName);
             for ($i = 0; $i < $len; ++$i) {
-                if (ctype_upper($propertyName[$i])) {
+                if (ctype_upper($propertyName[$i]) && $i !== 0) {
```

Comment:
This could be simplified with something like this, saving one call to `ctype_upper()`:
```php
if (0 !== $i && ($lcPropertyName = strtolower($propertyName[$i])) !== $propertyName[$i]) {
    $snakeCasedName .= '_'.$lcPropertyName;
}
```
BTW, if you don't agree with this approach, please move the cheaper check left.